training/views.py

```diff
@@ -0,0 +1,255 @@
+from training import models, forms
+
+from mailer import models as mailer_models
+from lib import xls
+from reports import forms
+
+from decorators import staff_required
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import redirect, render, get_object_or_404
+from django.core.urlresolvers import reverse
+from django.contrib import messages
+
+from datetime import datetime, timedelta
+
+@login_required
+def index(request):
+	'''Training index page. Lists events and links to add them'''
+
+	template = 'training/index.html'
+	context = {
+		'events': models.Event.objects.filter(date_time__gte=datetime.now()),
+		'user_events': models.Attendee.objects.filter(user=request.user).order_by('event__date_time')
+	}
+
+	return render(request, template, context)
+
+@login_required
+def view(request, event_id):
+	'''Displayes info on one event'''
+
+	event = get_object_or_404(models.Event, pk=event_id, date_time__gte=datetime.now())
+	attendee_check = models.Attendee.objects.filter(event=event, user=request.user)
+
+	template = 'training/view.html'
+	context = {
+		'event': event,
+		'attendee_check': attendee_check,
+	}
+
+	return render(request, template, context)
+
+@login_required
+def attend(request, event_id):
+	'''Adds the current user as an attendee to the given event'''
+
+	event = get_object_or_404(models.Event, pk=event_id, date_time__gte=datetime.now())
+	attendee = models.Attendee(event=event, user=request.user)
+	attendee.save()
+
+	return redirect(reverse('training_view', kwargs={'event_id': event_id}))
+
+@login_required
+def withdraw(request, event_id):
+	'''Drops the user from attendance sheet for this event, only if more than 24hrs before the event, notifies staff'''
+
+	event = get_object_or_404(models.Event, pk=event_id)
+	attendee = get_object_or_404(models.Attendee, event=event, user=request.user)
+
+	tf_hours = datetime.now() + timedelta(hours=24)
+
+	if tf_hours < event.date_time:
+		attendee.delete()
+		messages.add_message(request, messages.INFO, 'You have been withdrawn from this event.')
+	else:
+		messages.add_message(request, messages.INFO, 'This event is within the next 24 hours. You cannot withdraw.')
```

Comment:
Should we put a standard "contact administrator staff" message in here? Better that they email someone than that they don't show up.